samcli/local/lambda_service/local_lambda_invoke_service.py

```diff
@@ -71,17 +124,29 @@ def _invoke_request_handler(self, function_name):
 
         """
         flask_request = request
-        request_data = flask_request.get_data().decode('utf-8')
+
+        request_data = flask_request.get_data()
+
+        if not request_data:
+            request_data = b'{}'
+
+        request_data = request_data.decode('utf-8')
 
         stdout_stream = io.BytesIO()
 
         try:
             self.lambda_runner.invoke(function_name, request_data, stdout=stdout_stream, stderr=self.stderr)
         except FunctionNotFound:
-            # TODO Change this
-            raise Exception('Change this later')
+            LOG.debug('%s was not found to invoke.', function_name)
+            return LambdaErrorResponses.resource_not_found(function_name)
+
+        lambda_response, lambda_logs, is_lambda_user_error_response = \
+            LambdaOutputParser.get_lambda_output(stdout_stream)
 
-        lambda_response, lambda_logs = LambdaOutputParser.get_lambda_output(stdout_stream)
+        if is_lambda_user_error_response:
+            return self._service_response(lambda_response,
```

Comment:
This is the response I got from Lambda Service on a sample invoke that failed. We are missing `x-amzn-requestid`, `date`, `x-amz-executed-version` from these. Can we add them?

In fact, can we add a requestId header to every invocation?

```json
{
  "date": "Thu, 05 Jul 2018 20:24:47 GMT",
  "content-type": "application/json",
  "content-length": "114",
  "connection": "keep-alive",
  "x-amzn-requestid": "754e50c3-8091-11e8-afcc-a3b411a0cfac",
  "x-amz-function-error": "Unhandled",
  "x-amzn-remapped-content-length": "0",
  "x-amz-executed-version": "$LATEST",
  "x-amzn-trace-id": "root=1-5b3e7e8e-81e41598a657a40c4ad823a6;sampled=0"
}
```

Reply:
The lambda docker does not allow any requestId setting (it generates the uuid on the fly). So in order to make this experience clean, we would have to parse the output of the logs for the requestId and then set the header. That seems overkill. As of know, I am going to omit the requestId header.

* Adding `x-amz-executed-version` makes sense to me and will default to `$LATEST` (since we do not support qualifiers). 
* Not sure what `x-amzn-remapped-content-length` means or how it is commuted, so was going to omit it.
* `x-amzn-trace-id` only makes sense if we support XRay right?

Reply:
Okay, let's omit requestId. How about setting `x-amz-executed-version` to `$LATEST` because there is no version locally?

Following up, what if they use `AutoPublishAlias` and try to invoke the alias? I am okay if we don't support aliases now. Doesn't hurt.

Reply:
From my response :):
>Adding x-amz-executed-version makes sense to me and will default to $LATEST (since we do not support qualifiers).

Qualifiers might be easy to just add but we don't have a concept of a qualifier locally. I think I will hold off on adding the qualifier for now.